openmls/src/group/tests_and_kats/tests/proposal_validation.rs

```diff
@@ -1077,13 +1073,6 @@ fn test_valsem105() {
             KeyPackageTestVersion::WrongVersion => {
                 franken_key_package.protocol_version = 999;
             }
```

Comment:
I'm not sure I understand why these tests need to go.

Reply:
Ah, I think I misunderstood what they were testing. Let me take another look.

Reply:
Ah, no I think I removed these because the RFC doesn't actually require that the version and ciphersuite from the leaf node match those of the group, or the key package containing it. I suppose that is a bug in the RFC?

Reply:
The RFC has this verification of key packages
"Verify that the cipher suite and protocol version of the KeyPackage match those in the GroupContext."

https://www.rfc-editor.org/rfc/rfc9420.html#section-10.1-4.1

Some checks here look like they are for the capabilities though?
And the ciphersuite needs to be supported by everyone.

So I think these tests should still be valid. Or do I misunderstand what they are doing?

Reply:
The checks you reference are for `key_package.cipher_suite` and `key_package.protocol_version`. The tests I remove check what happens for `key_package.leaf_node.capabilities.cipher_suites` and `key_package.leaf_node.capabilities.protocol_versions`.

These are two different things, and the RFC fails to actually bring them into any kind of relation.